modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/CursorResetInterceptor.java

```diff
@@ -72,4 +68,28 @@ public Throwable onError(ExecutionContext<OperationModel> executionContext, Thro
 
     return exception;
   }
+
+  private List<Cursor> getParameterCursors(ExecutionContext<OperationModel> ctx) {
```

Comment:
this introduces a performance regression. With this code, the list of parameters is iterated on every single operation execution, when the reality is that the stream parmeters are always the same. This is why the current implementation receives the exact list of parameter names to be fetched. This solution must respect that. This interceptor must have direct access to the parameters it needs to fetch

Reply:
Oh, let me explain because it might not be clear, but this solution is similar to the existing one, where the stream parameters are pre-filtered.
Now, the constructor of CursorResetInterceptor  class receives as an argument the map with the stream parameters already filtered (done on ComponentMessageProcessor). Then, the 'before' method uses this pre-filtered Map and iterates it to fetch the value of each of the stream parameters from the context. It only iterates that map, not all the parameters. It is similar to the previous approach, but instead of using a list now it uses a map.